### PR TITLE
Create authorization and authentication whit jwt and controller, repository and model for roles

### PR DESCRIPTION
**Pull request by boro0208**
Create authentication and authorization for login and access API-s. Create a model, controller, and repository for Roles. Some files are written from zero.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -29,6 +29,25 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.flywaydb</groupId>
+            <artifactId>flyway-core</artifactId>
+            <version>8.5.12</version>
+        </dependency>
+        <dependency>
+            <groupId>org.flywaydb</groupId>
```
Comment (vajda) [inline pom.xml:38]: I think this dependency is not needed.

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
+spring.flyway.baselineOnMigrate = true
+spring.datasource.testWhileIdle = true
+spring.datasource.timeBetweenEvictionRunsMillis = 60000
+spring.datasource.minEvictableIdleTimeMillis = 30000
+spring.datasource.validationQuery = SELECT 1
+spring.datasource.max-active = 15
+spring.datasource.max-idle = 10
+spring.datasource.max-wait = 8000
+
+flyway.url = jdbc:mysql://localhost:3306/mysql
```
Comment (vajda) [inline src/main/resources/application.properties:17]: I think those `flyway.*` properties are no necessary. Flyway should be able to pick up the values from `spring.datasource`. And those are already defined there.

File: pom.xml (hunk 1 above)
Comment (boro0208) [reply pom.xml]: this dependency is unnecessary and I have removed it

File: src/main/resources/application.properties (hunk 2 above)
Comment (boro0208) [reply src/main/resources/application.properties:17]: I removed unnecessary properties

**Pull request merged**